Add formulae for portable-ruby 2.3 and 2.4

mistydemeo opened the pull request:
These were broken due to an [upstream bug](https://bugs.ruby-lang.org/issues/13413) in makefile generation, which prevented the `.rb` portions of statically linked extensions from being installed. This has been fixed in trunk, so I've backported the relevant patch in these formulae and put in a request for it to be backported into the branches upstream.

Trunk, and by extension the upcoming 2.5, is still broken in statically-linked builds for unrelated reasons. I'm looking into the issue.

Having these work will be useful for us since we want to start shipping a vendored Ruby 2.3 when 10.13 is out. :)

reitermarkus commented:
<a href="#" title="+1"><img src="https://camo.githubusercontent.com/3c156d4160d3377458b42b2a4c3754324e3d07f8/68747470733a2f2f7777772e656d6f6a69626173652e636f6d2f7265736f75726365732f696d672f656d6f6a69732f6170706c652f7831663434642e706e672e7061676573706565642e69632e715832503850375f67312e706e67" height="64px"></a>

xu-cheng commented:
Should we remove old rubies and only keep ruby 2,3?

Otherwise, :+1:

xu-cheng commented on Formula/portable-ruby@2.3.rb:
```diff
@@ -0,0 +1,135 @@
+
+  # Fixes the static build: https://bugs.ruby-lang.org/issues/13413
+  patch do
+    url "https://github.com/ruby/ruby/commit/b3dbeb6e90f316584f70e33f6bfb9d83fa5f30d3.patch?full_index=1"
```
Do we need `?full_index=1`

mistydemeo replied:
Yeah, this is a transitional feature for raw patch URLs that uses unabbreviated git refs instead of shortened ones.

MikeMcQuaid commented:
Great work figuring out the bug here 👏 

> portable-ruby@2.3 2.3.4

I think we should match Apple's version here: `ruby 2.3.3p222 (2016-11-21 revision 56859)`.

> portable-ruby@2.4 2.4.1

Where are we planning on using this formula?

mistydemeo commented:
> I think we should match Apple's version here: ruby 2.3.3p222 (2016-11-21 revision 56859).

👍 

> Where are we planning on using this formula?

It's mainly here for us to experiment with. Given the issues we've seen with static builds in the past, I also think it's good for us to test newer Ruby versions by having formulae for them here, even if we're not using them yet.

MikeMcQuaid commented:
> It's mainly here for us to experiment with. Given the issues we've seen with static builds in the past, I also think it's good for us to test newer Ruby versions by having formulae for them here, even if we're not using them yet.

Cool with me, just curious 👍. I think at some point it would be worth separating this repo into the stuff we're using/supporting in production and the stuff we aren't (but that's not for this PR).

mistydemeo commented:
Updated to switch portable-ruby@2.3 to use Ruby 2.3.3 - that point release is the same patchlevel as macOS 10.13's version. Also updated the comment on 2.3's patch to indicate that the commit has been backported onto the 2.3 branch.
